### PR TITLE
fix: display profile header on public page

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1645

- Display profile on the public account page
- Use avatar from profile in the header
- fix a race condition in Apollo client init

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. Default public profile: https://deploy-preview-1857--regen-registry.netlify.app/ecocredits/accounts/regen1typewgr0qksjl0m2r0cal530zl0g835wsunxna/portfolio
2. Shared account public profile: https://deploy-preview-1857--regen-registry.netlify.app/ecocredits/accounts/regen1df675r9vnf7pdedn4sf26svdsem3ugavgxmy46/portfolio

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@wgwz Here's the fix for adding the profile header to the public page. For some reason, the `PartyByAddr` queries are not firing on the preview (they're working fine locally), any idea why ?
/cc @blushi

**Comment by blushi** [comment]
> @wgwz Here's the fix for adding the profile header to the public page. For some reason, the `PartyByAddr` queries are not triggered on the preview (they're working fine locally), any idea why ? /cc @blushi

maybe something to do with caching?

**Comment by flagrede** [comment]
@blushi doesn't seem to be cache related, from react-query devtools the requests seems to be fetching indefinitly.

File: web-registry/src/apollo.tsx [hunk 1]
```diff
@@ -29,19 +29,12 @@ export const AuthApolloProvider = ({
...
-          });
-        }
-      }),
-  );
+  const authLink = setContext((_, { headers }) => ({
+    headers: {
+      ...headers,
+      'X-CSRF-TOKEN': query.data,
+    },
+  }));
```
Comment (wgwz) [inline web-registry/src/apollo.tsx:37]: 👍

**Review by wgwz — APPROVED** [review]
tACK and LGTM! sorry for the bug 😬 nice work figuring it out, that was a hard one.

Screen Shot 2023-04-06 at 11 11 18 AM

**Pull request merged**